### PR TITLE
Microcontroller programmed to control tripod

### PR DESCRIPTION
**Pull request by StimkaT**

### CODE REVIEW

File: electron/messagesHandlerFromWeb.js [hunk 1]
```diff
@@ -29,12 +29,24 @@ async  function messagesHandlerFromWeb(json, note) {
...
 
-        if (messageBuffers[data.deviceName].endsWith('\n')) {
+        const messageStr = message.toString()
+          .replace(/\n/g, '')
+          .replace(/\r/g, '')
+          .replace(/\x03/g, '');
+        messageBuffers[data.deviceName] += messageStr;
+        
+        if (message.toString().endsWith('\r\n')) {
+          console.log(111111111);
```
Comment (StimkaT) [inline electron/messagesHandlerFromWeb.js:48]: dell

File: projects/app/src/lib/services/messagesFromElectron.service.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ export class MessagesFromElectronService {
     } else if (message.event === 'USB_DEVICES_PORT_IS_CLOSED') {
       this.store.dispatch(setClosePort({ deviceName: message.data.deviceName }));
     } else if (message.event === 'FROM_USB_DEVICE') {
+      // console.log('1111', message)
```
Comment (StimkaT) [inline projects/app/src/lib/services/messagesFromElectron.service.ts:26]: delete

File: projects/tripod/src/lib/containers/control-buttons-container/control-buttons-container.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<page-layout>
+  <nav-panel-container toplBar></nav-panel-container>
+  <div body>
+    <control-buttons (emitter)="events($event)"></control-buttons>
+    <!-- [mode]="getMode$ | async" [led]="getLed$ | async"  -->
```
Comment (StimkaT) [inline projects/tripod/src/lib/containers/control-buttons-container/control-buttons-container.html:5]: delete

File: projects/tripod/src/lib/containers/control-buttons-container/control-buttons-container.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    NavPanelContainer,
+    RouterOutlet,
+    AsyncPipe,
+  ],
+  templateUrl: './control-buttons-container.html',
+  styles: ``,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ControlButtonsContainer {
+  // getMode$: Observable<string | null> = this.store.select(getMode);
```
Comment (StimkaT) [inline projects/tripod/src/lib/containers/control-buttons-container/control-buttons-container.ts:26]: delete

File: projects/tripod/src/lib/containers/control-video-container/control-video-container.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<page-layout>
+  <nav-panel-container toplBar></nav-panel-container>
+  <div body>
+
+    <img class="qqq" src="./assets/images/blink/ON.png" alt="Video">
+    <control-buttons (emitter)="events($event)"></control-buttons>
+    <!-- [mode]="getMode$ | async" [led]="getLed$ | async"  -->
```
Comment (StimkaT) [inline projects/tripod/src/lib/containers/control-video-container/control-video-container.html:7]: delete

File: projects/tripod/src/lib/containers/control-video-container/control-video-container.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    RouterOutlet,
+    AsyncPipe,
+  ],
+  templateUrl: './control-video-container.html',
+  styleUrl: './control-video-container.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ControlVideoContainer {
+  // getMode$: Observable<string | null> = this.store.select(getMode);
+  // getLed$: Observable<string | null> = this.store.select(getLed);
```
Comment (StimkaT) [inline projects/tripod/src/lib/containers/control-video-container/control-video-container.ts:29]: delete

File: electron/messagesHandlerFromWeb.js (hunk 1 above)
Comment (sislex) [reply electron/messagesHandlerFromWeb.js:48]: console.log('');
        console.log(message);
        console.log(message.toString().trim());
        console.log(message.toString().length);
        console.log('');

remove it

Comment (sislex) [reply electron/messagesHandlerFromWeb.js:48]: console.log(messageBuffers[data.deviceName].trim());  remove it

File: projects/tripod/arduino/widget-tripod/widget-tripod.ino [hunk 7]
```diff
@@ -89,19 +132,18 @@ void events(const String message) {
...
-    const char* command = doc["data"]["command"];
 
-    if (strcmp(event, "LED") == 0) {
-      info.mode = command;
-      if (info.mode.equals("ON")) {
-        info.led = "ON";
-        digitalWrite(ledPin, info.led.equals("ON") ? HIGH : LOW);
+    if (strcmp(event, "MOVE") == 0) {
+      const int steps1 = doc["data"]["steps1"];
+      const int s2 = doc["data"]["steps2"];
```
Comment (sislex) [inline projects/tripod/arduino/widget-tripod/widget-tripod.ino:138]: rename s2 to steps2

File: projects/tripod/arduino/widget-tripod/widget-tripod.ino [hunk 8]
```diff
@@ -89,19 +132,18 @@ void events(const String message) {
...
+      if (steps1 != 0) {
+        info.steps1 += steps1;
         sendDeviceInfo(timestamp);
-      } else if (info.mode.equals("OFF")) {
-        info.led = "OFF";
-        digitalWrite(ledPin, info.led.equals("ON") ? HIGH : LOW);
+      } 
+      if (s2 != 0) {
```
Comment (sislex) [inline projects/tripod/arduino/widget-tripod/widget-tripod.ino:143]: rename s2 to steps2

**Pull request merged**